gdal/frmts/grib/gribdataset.h

```diff
@@ -178,28 +190,19 @@ class GRIBRasterBand final: public GDALPamRasterBand
     std::string m_osType{};
     int     m_nPDTN = -1;
     std::vector<GUInt32> m_anPDSTemplateAssembledValues{};
+    bool    bLoadedPDS = false;
+    bool    bLoadedMetadata = false;
 };
 
 namespace gdal {
 namespace grib {
 
 // Thin layer to manage allocation and deallocation.
-class InventoryWrapper {
+class InventoryWrapper
+{
   public:
-
-    explicit InventoryWrapper(VSILFILE * fp)
-        : inv_(nullptr), inv_len_(0), num_messages_(0), result_(0) {
-      result_ = GRIB2Inventory(fp, &inv_, &inv_len_,
-                               0 /* all messages */, &num_messages_);
-    }
-
-    ~InventoryWrapper() {
-        if (inv_ == nullptr) return;
-        for (uInt4 i = 0; i < inv_len_; i++) {
-            GRIB2InventoryFree(inv_ + i);
-        }
-        free(inv_);
-    }
+    InventoryWrapper() : inv_(nullptr), inv_len_(0), num_messages_(0), result_(0) {}
```

Comment:
please use inline C++11 initialization of the members. That makes it more obvious that all are initialized